doc/examples/uuid.rst

```diff
@@ -164,11 +165,48 @@ In this example, round-tripping the document using the incorrect
 ``PYTHON_LEGACY``) changes the :class:`~bson.binary.Binary` subtype as a
 side-effect. **Note that this can also happen when the situation is reversed -
 i.e. when the original document is written using ``STANDARD`` representation
-and then round-tripped using the ``PYTHON_LEGACY`` representation. Note also
-that replacing ``PYTHON_LEGACY`` by ``JAVA_LEGACY`` or ``CSHARP_LEGACY`` in
-the above example produces the same behavior.**
+and then round-tripped using the ``PYTHON_LEGACY`` representation.**
 
-.. note:: Starting in PyMongo 4.0, this issue will be resolved as
+In the next example, we see the consequences of incorrectly using a
+representation that modifies byte-order (``CSHARP_LEGACY`` or ``JAVA_LEGACY``)
+when round-tripping documents::
+
+  from bson.codec_options import CodecOptions, DEFAULT_CODEC_OPTIONS
+  from bson.binary import Binary, UuidRepresentation
+  from uuid import uuid4
+
+  # Using UuidRepresentation.STANDARD stores a Binary subtype-4 UUID
+  std_opts = CodecOptions(uuid_representation=UuidRepresentation.STANDARD)
+  input_uuid = uuid4()
+  collection = client.testdb.get_collection('test', codec_options=std_opts)
+  collection.insert_one({'_id': 'baz', 'uuid': input_uuid})
+  assert collection.find_one({'uuid': Binary(input_uuid.bytes, 4)})['_id'] == 'baz'
+
+  # Retrieving this document using UuidRepresentation.JAVA_LEGACY returns a native UUID
+  # without modifying the UUID byte-order
+  java_opts = CodecOptions(uuid_representation=UuidRepresentation.JAVA_LEGACY)
+  java_collection = client.testdb.get_collection('test', codec_options=java_opts)
+  doc = java_collection.find_one({'_id': 'baz'})
+  assert doc['uuid'] == input_uuid
+
+  # Round-tripping the retrieved document silently changes the Binary bytes and subtype
+  java_collection.replace_one({'_id': 'baz'}, doc)
+  assert collection.find_one({'uuid': Binary(input_uuid.bytes, 3)}) is None
+  round_tripped_doc = collection.find_one({'_id': 'baz'})
+  assert round_tripped_doc['uuid'] == Binary(input_uuid.bytes, 3).as_uuid(UuidRepresentation.JAVA_LEGACY)
+
+
+In this case, using the incorrect :class:`~bson.binary.UuidRepresentation`
+(``JAVA_LEGACY`` instead of ``STANDARD``) changes the
+:class:`~bson.binary.Binary` bytes and subtype as a side-effect.
+**Note that this happens when any representation that
+manipulates byte-order (``CSHARP_LEGACY`` or ``JAVA_LEGACY``) is incorrectly
+used to round-trip UUIDs written with ``STANDARD``. When the situation is
+reversed - i.e. when the original document is written using ``STANDARD``
+and then round-tripped using ``CSHARP_LEGACY`` or ``JAVA_LEGACY`` -
+only the :class:`~bson.binary.Binary` subtype is changed.**
```

Comment:
The reverse situation would also change the bytes stored in the database. Also I think the "i.e. when the original document is written using ``STANDARD``..." part is written backwards.

Reply:
You are right about it being backwards. 
However, I think that when the original UUID is stored with *_LEGACY and then round-tripped using STANDARD only the subtype would change - STANDARD reads subtype 3 as PYTHON_LEGACY and writes them back as subtype 4 without changing byte-order.

Reply:
Ah right. So the bytes on the server don't change, but the python UUID instance will be different.